### ISSUE TITLE
Boxes identified from one window are being seen on other window.

### ISSUE THREAD

**Issue by sarvaatmak** [mikel-brostrom/Yolov5_DeepSort_Pytorch#297]
When i give two streams to track.py I am shown two two windows and detection from one window is seen on second window though there is no object on second window and vice versa. Take a look at given image. Ihave not found any problem with code for single stream it works fine but creates problem with multistreams also detections are also correct.
![Screenshot from 2022-02-10 18-02-19](https://user-images.githubusercontent.com/63709931/153409453-89aa08fe-2fe0-418d-83b3-a477945d630c.png)

**Comment by mikel-brostrom** [comment]
Check: https://github.com/mikel-brostrom/Yolov5_DeepSort_Pytorch/issues/293#issuecomment-1027641698

**Comment by sarvaatmak** [comment]
# According to the comment I need to run this command and then again run track.py but i get the following error please correct me, what I did is as follows.
yolo$ git checkout -b multi-video-tracking
yolo$ git checkout multi-video-tracking
yolo$ git add streams.txt

# so there is only one file i.e streams.txt in branch multi-video-tracker

but still i see the same faulty result what am i doing wrong?

**Comment by mikel-brostrom** [comment]
> According to the comment I need to create branch

No, read the comment again please

**Comment by sarvaatmak** [comment]
> No, read the comment again please

Do you mean to say just run command git checkout multi-video-tracking.
But it gave me this error.
error: pathspec ' multi-video-tracking' did not match any file(s) known to git

so I diid 


> # According to the comment I need to create branch and add streams .txt to that file and run this command and then again run track.py but I get the following error, what I did is as follows.
> yolo$ git checkout -b multi-video-tracking yolo$ git checkout multi-video-tracking yolo$ git add streams.txt
> 
> # so there is only one file i.e streams.txt in branch multi-video-tracker
> but still i see the same faulty result what am i doing wrong?

please help me sort this out  I am really confused.

**Comment by sarvaatmak** [comment]
> > According to the comment I need to create branch
> 
> No, read the comment again please

yes the problem is solved I had not done --recurse-submodules because i was not aware of this thanks.
Also can you please explain why this is necessary I want to know because say I want to create framework and then copy all folders to some device say nvidia jetson or raspberry pi then will simply copying this folder will work after all requirements are satisfied, or there also same problem will arise now I am working on GPU device and I am asking this keeping in mind future developments that I may perform.

**Comment by liang-jingyi** [comment]
@mikel-brostrom @sarvaatmak How to display ID, class and confidence on the output video? My output video only shows ID. Please help me

**Comment by sarvaatmak** [comment]
> @mikel-brostrom @sarvaatmak How to display ID, class and confidence on the output video? My output video only shows ID. Please help me

In track.py check what is passed into annotator.box_label

      annotator.box_label(bboxes, label, color=colors(c, True))
if you look at the code label contains id, class name and confidence in your case it may be providing only id.
